src/riak_kv_ttaaefs_manager.erl

```diff
@@ -345,53 +385,146 @@ handle_cast({all_sync, ReqID, From, _Now}, State) ->
-                                NextBucketList, partial, State),
+                                NextBucketList, partial, State,
+                                day_check),
             {noreply, State0, Timeout}
     end;
-handle_cast({day_sync, ReqID, From, Now}, State) ->
+handle_cast({hour_check, ReqID, From, Now}, State) ->
+    {MegaSecs, Secs, _MicroSecs} = Now,
+    UpperTime = MegaSecs * ?MEGA  + Secs,
+    LowerTime = UpperTime - 60 * 60,
     case State#state.scope of
         all ->
-            lager:warning("Invalid work_item=day_sync for Scope=all"),
-            {noreply, State, ?INITIAL_TIMEOUT};
+            Filter =
+                {filter, all, all, large, all,
+                {LowerTime, UpperTime}, pre_hash},
+            {State0, Timeout} =
+                sync_clusters(From,
+                                ReqID,
+                                State#state.local_nval,
+                                State#state.remote_nval,
+                                Filter,
+                                undefined, 
+                                full,
+                                State,
+                                hour_check),
+            {noreply, State0, Timeout};
         bucket ->
             [H|T] = State#state.bucket_list,
-            {MegaSecs, Secs, _MicroSecs} = Now,
-            UpperTime = MegaSecs * 1000000  + Secs,
-            LowerTime = UpperTime - 60 * 60 * 24,
+            
             % Note that the tree size is amended as well as the time range.
             % The bigger the time range, the bigger the tree.  Bigger trees
             % are less efficient when there is little change, but can more
-            % accurately reflect bigger changes (with less fasle positives).
+            % accurately reflect bigger changes (with less false positives).
             Filter =
-                {filter, H, all, medium, all,
+                {filter, H, all, small, all,
                 {LowerTime, UpperTime}, pre_hash},
             NextBucketList = T ++ [H],
             {State0, Timeout} =
                 sync_clusters(From, ReqID, range, range, Filter,
-                                NextBucketList, partial, State),
+                                NextBucketList, partial, State,
+                                hour_check),
             {noreply, State0, Timeout}
+    end;
+handle_cast({range_check, ReqID, From, Now}, State) ->
+    Range =
+        case get_range() of
+            none ->
+                case State#state.previous_success of
+                    false ->
+                        lager:info("No range sync as no range set"),
+                        none;
+                    {PrevMega, PrevSecs, _PrevMS} ->
+                        {MegaSecs, Secs, _MicroSecs} = Now,
+                        NowSecs = MegaSecs * ?MEGA  + Secs,
+                        {all,
+                            all,
+                            PrevMega * ?MEGA + PrevSecs - 60,
```

Comment:
This confuses me. In the hourly and daily, the Now argument is reduced by nr of seconds in day/hour respectively.
But here we subtract seconds from the previous successful sync. Should we not have something like, between max(PreviousSuccess , Now - 60), Now ?

Reply:
The 60 is a bit of a fuzz factor, that part-fixes a problem that the last success was prompted potentially a bit before this timestamp.  I'll either get the correct timestamp passed through in some way, or make it replace 60 with ?FUZZ with a comment on the definition of FUZZ to explain

Reply:
Actually, no, I was wrong.  The last success time is the time the last successful query was prompted.  So I think the fuzz factor is unnecessary